lib/maths/CSignal.cc

```diff
@@ -910,6 +910,38 @@ std::size_t CSignal::selectComponentSize(const TFloatMeanAccumulatorVec& values,
     return size;
 }
 
+CSignal::TMeanAccumulatorVec
+CSignal::smoothResample(std::size_t size, TMeanAccumulatorVec component) {
+
+    if (size >= component.size()) {
```

Comment:
Is it worth adding `|| size == 0` to the condition here to defend against division by zero later on if this method is ever reused in the future in a place where `size` could be zero?

Reply:
I think it is probably worth both. We can use a best guess which is `size == 1`, but it isn't really clear what the user intention is so I think it is worth documenting that it shouldn't be called like this. I made these changes in [e45d65d](https://github.com/elastic/ml-cpp/pull/1604/commits/e45d65d0e3f08c31bfdc69d3bbebe8e486f14a12).